deploy/runtools/run_farm.py

```diff
@@ -364,537 +427,163 @@ def terminate_run_farm(self, terminatesomef1_16: int, terminatesomef1_4: int, te
-                # we're done AND we have switches running here, so kill them,
-                # then copy off their logs. this handles the case where you
-                # have a node with one simulation and some switches, to make
-                # sure the switch logs are copied off.
-                #
-                # the other cases are when you have multiple sims and a cycle-acc network,
-                # in which case the all() will never actually happen (unless someone builds
-                # a workload where two sims exit at exactly the same time, which we should
-                # advise users not to do)
-                #
-                # a last use case is when there's no network, in which case
-                # instance_assigned_switches won't be true, so this won't be called
-
-                self.kill_switches_instance()
-
-                for counter, switchsim in enumerate(self.parentnode.switch_slots):
-                    switchsim.copy_back_switchlog_from_run(job_results_dir, counter)
-
-            if now_done and terminateoncompletion:
-                # terminate the instance since everything is done and instance
-                # termination is enabled
-                assert isinstance(self.parentnode.boto3_instance_object, EC2InstanceResource)
-                instanceids = get_instance_ids_for_instances([self.parentnode.boto3_instance_object])
-                terminate_instances(instanceids, dryrun=False)
-
-            return {'switches': switchescompleteddict, 'sims': jobs_done_q}
-
-        assert False, "Instance must host switch slots and/or FPGA slots"
+        # then iterate through them
+        for max_simcount, sim_host_handle in self.SORTED_SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS:
+            if max_simcount < num_sims:
+                # instance doesn't support enough sims
+                continue
+            consumed = self.mapper_consumed[sim_host_handle]
+            if consumed:
+                # instance supports enough sims but none are available
+                continue
+            return sim_host_handle
+
+        rootLogger.critical(f"ERROR: No run hosts are available to satisfy the request for an instance with support for {num_sims} simulation slots. Add more instances in your runfarm configuration (e.g., config_run_farm.ini).")
+        raise Exception
+
+    def allocate_sim_host(self, sim_host_handle: str) -> Inst:
+        """ Let user allocate and use an instance (assign sims, etc.).
+        This deliberately exposes sim_host_handles to users, so that if
+        they know exactly how to map to a particular platform, they always
+        have an escape hatch. """
+        inst_ret = self.run_farm_hosts_dict[sim_host_handle]
+        assert self.mapper_consumed[sim_host_handle] == False, "{sim_host_handle} is already allocated."
+        self.mapper_consumed[sim_host_handle] = True
+        return inst_ret
+
+    def get_default_switch_host_handle(self) -> str:
```

Comment:
I don't support networked sims (mainly because I never tested them - I figure this can be punted to later). The logic in this function shouldn't be used since networked sims are unsupported, but the logic here is sufficient for a basic networked sim case (though it can be improved).